### PR TITLE
Clean up unsed MHA code to avoid confusion

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105956

cc @ezyang @gchanan

### CODE REVIEW

File: docs/source/distributed.tensor.parallel.rst [hunk 1]
```diff
@@ -53,13 +53,11 @@ used for input/output preparation:
 .. autofunction::  make_output_shard_1d
 .. autofunction::  make_output_tensor
 
-Currently, there are some constraints which makes it hard for the `nn.MultiheadAttention`
-module to work out of box for Tensor Parallelism, so we built this multihead_attention
-module for Tensor Parallelism users. Also, in ``parallelize_module``, we automatically
-swap ``nn.MultiheadAttention`` to this custom module when specifying ``PairwiseParallel``.
+Currently, there are some constraints which makes it hard for the MultiheadAttention
+module to work out of box for Tensor Parallelism, so we recommend users to try `ColwiseParallel`
```
Comment (awgu) [inline docs/source/distributed.tensor.parallel.rst:57]: nit: Do these need to use double backtick (\`\`)? For in-code docs, I think single backtick becomes italics, and only double backtick becomes code-type. I am not sure for this `.rst` file.

**Review by wz337 — APPROVED** [review]
LGTM!

**Review by ezyang — APPROVED** [review]

**Review by wanchaol — APPROVED** [review]
Thanks for cleaning up!

**Comment by HMellor** [comment]
Hi, does this PR mean that the statement

> PairwiseParallel only supports nn.Multihead Attention, nn.Transformer or even-number-layer MLP for now.

from https://pytorch.org/docs/main/distributed.tensor.parallel.html#torch.distributed.tensor.parallel.style.PairwiseParallel is no longer correct?

**Comment by fduwjj** [comment]
@HMellor Oh, thanks for catching that. I think down the road, we only recommend using `PairwiseParallel ` for MLP because for MHA/attention, there are some customization needed so we recommend users to try colwise + rowwise for now and we will think about how the UX look like down the road.

**Pull request closed without merging**